resources/assets/coffee/react/_components/beatmap-basic-stats.coffee

```diff
@@ -32,16 +32,28 @@ formatDuration = (value) ->
   else
     "#{m}:#{_.padStart s, 2, 0}"
 
+truncate = (number) ->
+  suffixes = ["", "k", "m", "b", "t"]
+  k = 1000
+  i = Math.floor(Math.log(number) / Math.log(k))
+  j = number / Math.pow(k, i)
+  formattedNumber = j.toFixed()
+
+  if k > number
+    "#{number}"
+  else
+    "#{formattedNumber}#{suffixes[i]}"
+
 
 export BeatmapBasicStats = ({beatmapset, beatmap}) ->
   div
     className: bn
     for stat in ['total_length', 'bpm', 'count_circles', 'count_sliders']
       value =
         if stat == 'bpm'
-          beatmapset.bpm
+          truncate beatmapset.bpm
         else
-          beatmap[stat]
+          truncate beatmap[stat]
```

Comment:
this breaks the following `value` formatting (notably for maps longer than 1000 seconds) .